src/Hl7.Fhir.Base/Rest/BaseFhirClient.cs

```diff
@@ -809,7 +876,7 @@ await ValidateResponse(responseMessage, expect, getSerializationEngine(), sugges
         // the full body of the altered resource.
         var noRealBody = LastBodyAsResource is null || (LastBodyAsResource is OperationOutcome && string.IsNullOrEmpty(LastBodyAsResource.Id));
         var shouldFetchFullRepresentation = noRealBody
-                                            && isPostOrPutOrPatch(request)
+                                            && (msg is not null ? isPostOrPutOrPatch(msg.Method) : isPostOrPutOrPatch(entryComponent!))
```

Comment:
entryComponent can be null, even if HttpResponseMessage isn't, you can't be sure, although we haven't implemented it. A future collaborator could.

I don't like this difference of behaviour based on the presence of parameters.